components/SingleArticlePage.jsx

```diff
@@ -1,50 +1,76 @@
-import { useState, useEffect } from "react"
-import { useParams } from "react-router-dom"
-import { fetchArticleById } from "../utils/api"
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticleById, fetchCommentsByArticleId } from "../utils/api";
+import CommentCard from "./CommentCard";
+export default function SingleArticlePage() {
+  const [singleArticle, setSingleArticle] = useState({});
+  const [comments, setComments] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const { article_id } = useParams();
 
-export default function SingleArticlePage(){
-    const [singleArticle, setSingleArticle] = useState({})
-    const [isLoading, setIsLoading] = useState(false)
-    const [isError, setIsError] = useState(false)
-    const { article_id} = useParams()
-
-    useEffect(() => {
-        if (article_id){
-        setIsLoading(true)
-        setIsError(false)
-        fetchArticleById(article_id)
+  useEffect(() => {
+    if (article_id) {
+      setIsLoading(true);
+      setIsError(false);
+      fetchArticleById(article_id)
         .then((response) => {
-            setSingleArticle(response)
-            setIsLoading(false)
-        }).catch((err) => {
-            setIsError(true)
-            setIsLoading(false)
-        })}
-    }, [article_id])
- 
-if(isError){
-    return <p>ERROR 404 - Content not found</p>
-}
-if (isLoading){
-    return <div>LOADING!</div>            
-}
+          setSingleArticle(response);
+          setIsLoading(false);
+        })
+        .catch((err) => {
+          setIsError(true);
+          setIsLoading(false);
+        });
+    }
+  }, [article_id]);
 
+  useEffect(() => {
+    if (article_id) {
+      fetchCommentsByArticleId(article_id)
+        .then((commentData) => {
+          setComments(commentData);
+        })
+        .catch((err) => {
+          setIsError(true);
+        });
+    }
+  }, [article_id]);
```

Comment:
If you have two useEffects with the same dependencies, it's a good indicator that you should extract them into their own components.
In this case, if I can't load the comments, I can't see the article either, as your catch block would make update the error state for the whole SingleArticle component. It also doesn't look very clean.
I suggest moving the comments into their own component and handling all the funcitonality / error handling there.